src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns login status of the user and a url to log in or out. */
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    UserService userService = UserServiceFactory.getUserService();
+    String urlToRedirectTo = "/signup-test.html";
+    String json;
+
+    if(userService.isUserLoggedIn()) {
+      String logoutUrl = userService.createLogoutURL(urlToRedirectTo);
+      json = "{ \"status\": true, \"url\": \"" + logoutUrl + "\" }";
+    }
```

Comment:
They Style Guide suggests that else goes on the same line as the preceding bracket: https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style

Reply:
+1